module/trie.c

```diff
@@ -842,7 +842,7 @@ int TrieInsert_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv,
     char *empty = "";
     char **temp = RedisModule_Calloc(nstrings, sizeof(char*));
     for (int i = 0; i < nstrings; i++) {
-        temp[i] = RedisModule_StringPtrLen(argv[2 + i], &dummy);
+        temp[i] = strdup(RedisModule_StringPtrLen(argv[2 + i], &dummy));
```

Comment:
These were the compilation warnings: ```
make[2]: Entering directory '/home/dawson/dev/cs220/project/spellcheck/api/lib/redis-tries/module'
gcc -I../RedisModulesSDK/ -fPIC -g -lc -lm -W -Wall -fno-common -ggdb -std=gnu99 -O2   -c -o trie.o trie.c
trie.c: In function ‘TrieInsert_RedisCommand’:
trie.c:828:5: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
     if (argc <= 2)
     ^~
trie.c:831:2: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  RedisModuleKey *key = RedisModule_OpenKey(ctx, argv[1],
  ^~~~~~~~~~~~~~
trie.c:845:17: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
         temp[i] = RedisModule_StringPtrLen(argv[2 + i], &dummy);
                 ^
trie.c: In function ‘TrieContains_RedisCommand’:
trie.c:892:18: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     char *temp = RedisModule_StringPtrLen(argv[2], &dummy);
                  ^~~~~~~~~~~~~~~~~~~~~~~~
trie.c: In function ‘TrieCompletions_RedisCommand’:
trie.c:917:18: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     char *temp = RedisModule_StringPtrLen(argv[2], &dummy);
                  ^~~~~~~~~~~~~~~~~~~~~~~~
trie.c: In function ‘TrieApproxMatch_RedisCommand’:
trie.c:952:18: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     char *temp = RedisModule_StringPtrLen(argv[2], &dummy);
                  ^~~~~~~~~~~~~~~~~~~~~~~~
ld -o trie.so trie.o -shared -Bsymbolic 

```

Reply:
Duplicating the string allows you to work with a non-`const` version of the string, making it so that you don't get the discarding `const` qualifier error.

Reply:
Do you ever need to free temp, then?

Reply:
Temp will need to be freed after the operation. I don't know if Dustin is that and I'm on mobile, sorry.

Reply:
I'll try to do that right now.